libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -3098,6 +3012,122 @@ string YulUtilFunctions::conversionFunction(Type const& _from, Type const& _to)
 	});
 }
 
+string YulUtilFunctions::copyStructToStorageFunction(StructType const& _from, StructType const& _to)
+{
+	solAssert(_to.dataStoredIn(DataLocation::Storage), "");
+	solAssert(_from.structDefinition() == _to.structDefinition(), "");
+
+	string functionName =
+		"copy_struct_to_storage_from_" +
+		_from.identifier() +
+		"_to_" +
+		_to.identifier();
+
+	return m_functionCollector.createFunction(functionName, [&]() {
+		Whiskers templ(R"(
+			function <functionName>(slot, value) {
+				<?fromStorage> if eq(slot, value) { leave } </fromStorage>
```

Comment:
This is shortcut you asked for

Reply:
Hm, it will not be possible to inline this now - maybe we should turn that into a wrapping `if` statement?